public/app/features/teams/TeamSettings.test.tsx

```diff
@@ -18,19 +23,36 @@ const setup = (propOverrides?: object) => {
 
   Object.assign(props, propOverrides);
 
-  const wrapper = shallow(<TeamSettings {...props} />);
-  const instance = wrapper.instance() as any;
-
-  return {
-    wrapper,
-    instance,
-  };
+  render(<TeamSettings {...props} />);
 };
 
-describe('Render', () => {
+describe('Team settings', () => {
   it('should render component', () => {
-    const { wrapper } = setup();
+    setup();
+
+    expect(screen.getByText('Team settings')).toBeInTheDocument();
+  });
+
+  it('should validate required fields', async () => {
+    const mockUpdate = jest.fn();
+    setup({ updateTeam: mockUpdate });
+    await userEvent.clear(screen.getByRole('textbox', { name: /Name/ }));
+    await userEvent.type(screen.getByLabelText(/Email/i), 'team@test.com');
+    // Submitting with userEvent doesn't work here
+    fireEvent.submit(screen.getByRole('button', { name: 'Update' }));
```

Comment:
are you saying `await userEvent.click()` doesn't work here? 🤔 would be interesting to find out why 😕

Reply:
Yeah, also tried setting it up with userEvent.setup(). Maybe it's smth with our form component since i was able to make it work in other components.